onnxruntime/core/graph/graph.cc

```diff
@@ -1992,7 +1994,19 @@ Node& Graph::AddNode(const NodeProto& node_proto,
     attributes[attr.name()] = attr;
   }
 
-  return AddNode(node_proto.name(),
+  size_t current_op_type_count;
+  const auto op_type = node_proto.op_type();
+  if (type_to_count_map.find(op_type) == type_to_count_map.end())
+    current_op_type_count = type_to_count_map[op_type] = 1;
+  else
+    current_op_type_count = ++type_to_count_map[op_type];
+
+  std::string node_name = node_proto.name();
+  if (node_name.empty())
+    node_name = GenerateNodeName("unnamed_" + op_type + "_" +
```

Comment:
Why do we need this? This name won't be visible on Netron any way when debugging.

Reply:
maybe not, but atleast it helps if we can visualize the serialized proto (not sure if this is possible) and the information here would atleast tell us that a specific op type (offset by the specific count) malfunctioned (as opposed to having no information at all)